### PR TITLE
build: Use caret dependencies for Rust-VMM crates

### PR DESCRIPTION
**Pull request by rbradford**
And also bump the crate versions as well as the dependency versions.

See: rust-vmm/community#136

Signed-off-by: Rob Bradford <robert.bradford@intel.com>

### CODE REVIEW

**Review by liuw — APPROVED** [review]

File: crates/vfio-ioctls/Cargo.toml [hunk 1]
```diff
@@ -21,11 +21,11 @@ mshv = ["mshv-ioctls", "mshv-bindings"]
...
-kvm-ioctls = { version = "~0", optional = true }
+kvm-bindings = { version = "0.6.0", optional = true }
+kvm-ioctls = { version = "0.12.0", optional = true }
 thiserror = ">=1.0"
-vfio-bindings = "~0"
-vm-memory = { version = ">=0.6", features = ["backend-mmap"] }
-vmm-sys-util = ">=0.8.0"
+vfio-bindings = { version = "0.4.0", path = "../vfio-bindings" }
+vm-memory = { version = "0.10.0", features = ["backend-mmap"] }
+vmm-sys-util = "=0.11.0"
```
[the remark this answers is not in the document]
Comment (rbradford) [reply crates/vfio-ioctls/Cargo.toml:29]: Yes, that's a typo. Thank you!

File: crates/vfio-ioctls/Cargo.toml [hunk 2]
```diff
@@ -21,11 +21,11 @@ mshv = ["mshv-ioctls", "mshv-bindings"]
 byteorder = ">=1.2.1"
 libc = ">=0.2.39"
```
[the remark this answers is not in the document]
Comment (rbradford) [reply crates/vfio-ioctls/Cargo.toml:22]: You mean to caret versions or just generally? I didn't end up doing that for vhost in the end as there were some toolchain issues with one of the dependencies.

[the remark this answers is not in the document]
Comment (rbradford) [reply crates/vfio-ioctls/Cargo.toml]: Okay, I switched to caret versions. But I didn't update dependencies like libc.

**Comment by liuw** [comment]
@andreeaflorescu there seems to be issues with buildkite.

**Comment by andreeaflorescu** [comment]
@liuw The rust-vmm-ci submodule needs to be updated to get rid of the timeout. See: https://github.com/rust-vmm/community/issues/137. Dependabot didn't yet pick up the latest rust-vmm-ci commit, so the options are:
- wait for dependabot to create a new PR and merge that one
- manually update rust-vmm-ci as specified here: https://github.com/rust-vmm/community/issues/137#issuecomment-1332365977

In the meantime, I retriggered the build for this tests, but this will be a recurrent problem until we update the rust-vmm-ci.

**Comment by andreeaflorescu** [comment]
It looks like there is a legit failure for the coverage, @rbradford can you update that as well?

```
AssertionError: Coverage is increased from 77.7 to 79.1. Please update the coverage in coverage_config_x86_64.json.
```

**Review by sboeuf — APPROVED** [review]

**Pull request merged**